plugins/outputs/mqtt/homie.go

```diff
@@ -0,0 +1,136 @@
+package mqtt
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+)
+
+var idRe = regexp.MustCompile(`([^a-z0-9]+)`)
+
+func (m *MQTT) collectHomieDeviceMessages(topic string, metric telegraf.Metric) ([]message, string, error) {
+	var messages []message
+
+	// Check if the device-id is already registered
+	if _, found := m.homieSeen[topic]; !found {
+		deviceName, err := m.homieDeviceNameGenerator.Generate(metric)
+		if err != nil {
+			return nil, "", fmt.Errorf("generating device name failed: %w", err)
+		}
+		messages = append(messages, message{topic + "/$homie", []byte("4.0")})
+		messages = append(messages, message{topic + "/$name", []byte(deviceName)})
+		messages = append(messages, message{topic + "/$state", []byte("ready")})
+		m.homieSeen[topic] = make(map[string]bool)
+		// TODO: Declare last will
```

Comment:
Is there a follow-up PR needed here? Or something we can do to document this TODO?